src/graphql/graphql-api.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API, if its allowed
+	 *
+	 * @access private
+	 */
+	public static function init() {
+		/**
+		 * Filter to enable/disable the graphQL API. By default, it is enabled.
+		 * 
+		 * @param bool $is_graphql_to_be_enabled Whether the graphQL API should be enabled or not.
+		 */
+		$is_graphql_to_be_enabled = apply_filters( 'vip_block_data_api__is_graphql_enabled', true );
```

Comment:
This `init` method is being called on load (at the end of this file, not hooked into WordPress `init` or another lifecycle hook). Therefore, plugin / theme code may not be able to add a filter before this is executed. I would move this inside the `register_types` method.

Reply:
Good point, I have moved this into the `register_types` now.